### PR TITLE
DOC-600 Field level changes for release v.4.31 to v.4.37.0

### PR DESCRIPTION
**Pull request by asimms41**
## Description

Resolves https://github.com/redpanda-data/documentation-private/issues/2744
Review deadline: 10th October

## Page previews

Added `content_md5` field to [aws_s3 output](https://deploy-preview-65--redpanda-connect.netlify.app/redpanda-connect/components/outputs/aws_s3/?tab=tabs-1-advanced)
Added `send_ack` field to [nats_input](https://deploy-preview-65--redpanda-connect.netlify.app/redpanda-connect/components/inputs/nats/#send_ack)
Added auth fields to [schema_registry input](https://deploy-preview-65--redpanda-connect.netlify.app/redpanda-connect/components/inputs/schema_registry/#oauth) and [schema_registry output](https://deploy-preview-65--redpanda-connect.netlify.app/redpanda-connect/components/outputs/schema_registry/#oauth-enabled)
Added `job_project` to [gcp_bigquery output](https://deploy-preview-65--redpanda-connect.netlify.app/redpanda-connect/components/outputs/gcp_bigquery/) and reworded field descriptions
Made specifying the `schema_registry` in the [redpanda_migrator_bundle input](https://deploy-preview-65--redpanda-connect.netlify.app/redpanda-connect/components/inputs/redpanda_migrator_bundle/) and [redpanda_migrator_bundle output](https://deploy-preview-65--redpanda-connect.netlify.app/redpanda-connect/components/outputs/redpanda_migrator_bundle/) optional
Added `replication_factor` and `replication_factor_override` to [kafka_migrator input](https://deploy-preview-65--redpanda-connect.netlify.app/redpanda-connect/components/inputs/redpanda_migrator/?tab=tabs-1-advanced#replication_factor_override) and [kafka_migrator output](https://deploy-preview-65--redpanda-connect.netlify.app/redpanda-connect/components/outputs/redpanda_migrator/#replication_factor_override)
Added `metadata_max_age` to [kafka_franz input](https://deploy-preview-65--redpanda-connect.netlify.app/redpanda-connect/components/inputs/kafka_franz/#metadata_max_age)
Added `metadata_max_age` to [redpanda_migrator input](https://deploy-preview-65--redpanda-connect.netlify.app/redpanda-connect/components/inputs/redpanda_migrator/#metadata_max_age)
Added version number (4.37.0) to [redpanda_migrator_bundle input](https://deploy-preview-65--redpanda-connect.netlify.app/redpanda-connect/components/inputs/redpanda_migrator_bundle/) and [redpanda_migrator_bundle output](https://deploy-preview-65--redpanda-connect.netlify.app/redpanda-connect/components/outputs/redpanda_migrator_bundle/)
Added `fetch_in_order` field to the [schema_registry input](https://deploy-preview-65--redpanda-connect.netlify.app/redpanda-connect/components/inputs/schema_registry/#fetch_in_order)
Added `vector` method to [Bloblang methods ](https://deploy-preview-65--redpanda-connect.netlify.app/redpanda-connect/guides/bloblang/methods/)

## Checks

- [ ] New feature
- [x] Content gap
- [ ] Support Follow-up
- [ ] Small fix (typos, links, copyedits, etc)

### CODE REVIEW

File: modules/components/pages/inputs/schema_registry.adoc [hunk 1]
```diff
@@ -301,4 +302,132 @@ Set `auto_replay_nacks` to `false` to delete rejected messages. Disabling auto r
 
 *Default*: `true`
 
+=== `oauth`
```
Comment (asimms41) [inline modules/components/pages/inputs/schema_registry.adoc:305]: Please check these fields descriptions as I have reworded slightly to make them more generic.

File: modules/components/pages/outputs/schema_registry.adoc [hunk 2]
```diff
@@ -285,9 +274,136 @@ password: ${KEY_PASSWORD}
 
 The maximum number of messages to have in flight at a given time. Increase this number to improve throughput.
 
-
 *Type*: `int`
 
 *Default*: `64`
 
+=== `oauth`
```
Comment (asimms41) [inline modules/components/pages/outputs/schema_registry.adoc:281]: Please check these field descriptions as I have reworded slightly to make them less generic.

File: modules/components/pages/outputs/aws_s3.adoc [hunk 3]
```diff
@@ -242,9 +243,15 @@ This field supports xref:configuration:interpolation.adoc#bloblang-queries[inter
...
-This field supports xref:configuration:interpolation.adoc#bloblang-queries[interpolation functions].
+The content language to set for each object. This field supports xref:configuration:interpolation.adoc#bloblang-queries[interpolation functions].
+
+*Type*: `string`
+
+*Default*: `""`
+
+=== `content_md5`
 
+The https://docs.aws.amazon.com/AmazonS3/latest/userguide/checking-object-integrity.html#checking-object-integrity-md5[content MD5^] to set for each object. This field supports xref:configuration:interpolation.adoc#bloblang-queries[interpolation functions].
```
[the remark this answers is not in the document]
Comment (mihaitodor) [reply modules/components/pages/outputs/aws_s3.adoc:254]: I think that makes sense

File: modules/components/pages/inputs/schema_registry.adoc (hunk 1 above)
Comment (mihaitodor) [reply modules/components/pages/inputs/schema_registry.adoc:305]: I think this is fine, but please note these fields are also exposed by other components such as https://docs.redpanda.com/redpanda-connect/components/inputs/http_client/#oauth. I'm not sure what's the best way to ensure such common fields don't end up with varying descriptions across the docs...

File: modules/components/pages/outputs/schema_registry.adoc (hunk 2 above)
Comment (mihaitodor) [reply modules/components/pages/outputs/schema_registry.adoc:281]: Same

**Review by mihaitodor — APPROVED** [review]
LGTM, just a few comments.

File: modules/components/pages/inputs/schema_registry.adoc (hunk 1 above)
Comment (asimms41) [reply modules/components/pages/inputs/schema_registry.adoc:305]: Raising a ticket to add a content that can be reused and conditioned across different components/connectors: https://github.com/redpanda-data/documentation-private/issues/2760

File: modules/components/pages/outputs/schema_registry.adoc (hunk 2 above)
Comment (asimms41) [reply modules/components/pages/outputs/schema_registry.adoc:281]: Added a ticket for this: https://github.com/redpanda-data/documentation-private/issues/2760

File: modules/components/pages/outputs/gcp_bigquery.adoc [hunk 4]
```diff
@@ -86,16 +86,16 @@ output:
 
 == Credentials
 
-By default Redpanda Connect will use a shared credentials file when connecting to GCP services. You can find out more in xref:guides:cloud/gcp.adoc[].
```
[the remark this answers is not in the document]
Comment (mihaitodor) [reply modules/components/pages/outputs/gcp_bigquery.adoc:89]: I think the `job_project` docs look good.

**Review by mihaitodor — APPROVED** [review]

**Review by Jeffail — APPROVED** [review]

File: modules/components/pages/inputs/nats.adoc [hunk 5]
```diff
@@ -88,7 +90,7 @@ NATS component, so that monitoring tools between NATS and Redpanda Connect can s
 
 == Authentication
 
-There are several components within Redpanda Connect which uses NATS services. You will find that each of these components
+There are several components within Redpanda Connect which use NATS services. You will find that each of these components
 support optional advanced authentication parameters for https://docs.nats.io/nats-server/configuration/securing_nats/auth_intro/nkey_auth[NKeys^]
 and https://docs.nats.io/using-nats/developer/connecting/creds[User Credentials^].
```
Comment (JakeSCahill) [inline modules/components/pages/inputs/nats.adoc:95]:
```suggestion
Several components in Redpanda Connect use NATS services. Each of these components
supports optional advanced authentication parameters for https://docs.nats.io/nats-server/configuration/securing_nats/auth_intro/nkey_auth[NKeys^]
and https://docs.nats.io/using-nats/developer/connecting/creds[User Credentials^].
```

File: modules/components/pages/inputs/redpanda_migrator.adoc [hunk 6]
```diff
@@ -705,6 +708,32 @@ The label of the `redpanda_migrator` output in which the currently selected topi
 
 *Type*: `string`
 
-*Default*: `"redpanda_migrator_output"`
+*Default*: `redpanda_migrator_output`
+
+=== `replication_factor_override`
+
+Use the value specified in the `replication_factor` field when creating topics.
```
Comment (JakeSCahill) [inline modules/components/pages/inputs/redpanda_migrator.adoc:715]:
```suggestion
Indicate whether to override the replication factor of the input topics when creating copies of them. The new replication factor must be specified in the <<replication-factor, `replication_factor` field>>.
```

@Jeffail or @mihaitodor I'm unsure why the input has this option. Does the input or the output create the topic copies?

Comment (JakeSCahill) [reply modules/components/pages/inputs/redpanda_migrator.adoc:715]: Also, I assume if this is `false` that the new topics inherit the replication factor of the original. Is that right?

File: modules/components/pages/inputs/redpanda_migrator.adoc [hunk 7]
```diff
@@ -705,6 +708,32 @@ The label of the `redpanda_migrator` output in which the currently selected topi
 
 *Type*: `string`
 
-*Default*: `"redpanda_migrator_output"`
+*Default*: `redpanda_migrator_output`
+
+=== `replication_factor_override`
```
Comment (JakeSCahill) [inline modules/components/pages/inputs/redpanda_migrator.adoc:713]: I'd move `output_resource`, `replication_factor_override`, and `replication_factor` higher up the hierarchy so it's closer to `topics`.  These are all related to how the connector interacts with topics but it's in the same place as low-level batching settings.

File: modules/components/pages/inputs/redpanda_migrator.adoc [hunk 8]
```diff
@@ -705,6 +708,32 @@ The label of the `redpanda_migrator` output in which the currently selected topi
...
+Replication factor for created topics. This is only used when `replication_factor_override` is set to `true`.
+
+*Type*: `int`
+
+*Default*: `3`
+
+
+=== `metadata_max_age`
+
+The maximum period of time after which metadata is refreshed.
```
Comment (JakeSCahill) [inline modules/components/pages/inputs/redpanda_migrator.adoc:732]: What's the significance of this option? We should provide more guidance on when to change this.

File: modules/components/pages/inputs/redpanda_migrator.adoc [hunk 9]
```diff
@@ -705,6 +708,32 @@ The label of the `redpanda_migrator` output in which the currently selected topi
...
+
+=== `replication_factor_override`
+
+Use the value specified in the `replication_factor` field when creating topics.
+
+*Type*: `bool`
+
+*Default*: `true`
+
+=== `replication_factor`
```
Comment (JakeSCahill) [inline modules/components/pages/inputs/redpanda_migrator.adoc:721]:
```suggestion
[[replication-factor]]
=== `replication_factor`
```

File: modules/components/pages/inputs/schema_registry.adoc [hunk 10]
```diff
@@ -111,6 +128,18 @@ Include only subjects which match the regular expression filter, or leave this f
 
 *Default*: `""`
 
+=== `fetch_in_order`
+
+Fetch all schemas from the schema registry service and sort them by ID. Set this value to `true` if you use schema references: schemas that refer to other schemas.
```
Comment (JakeSCahill) [inline modules/components/pages/inputs/schema_registry.adoc:133]:
```suggestion
Fetch all schemas from the schema registry service and sort them by ID. Set this value to `true` if you use schemas that refer to other schemas (schema references).
```

File: modules/components/pages/inputs/schema_registry.adoc [hunk 11]
```diff
@@ -301,4 +302,132 @@ Set `auto_replay_nacks` to `false` to delete rejected messages. Disabling auto r
...
+
+Whether to use JWT authentication in requests.
+
+*Type*: `bool`
+
+*Default*: `false`
+
+=== `jwt.private_key_file`
+
+A file in PEM format encoded via PKCS1 or PKCS8 as private key.
```
Comment (JakeSCahill) [inline modules/components/pages/inputs/schema_registry.adoc:403]:
```suggestion
A file in PEM format encoded using PKCS1 or PKCS8 as private key.
```

File: modules/components/pages/outputs/gcp_bigquery.adoc [hunk 12]
```diff
@@ -216,7 +234,7 @@ Causes values not matching the schema to be tolerated. Unknown values are ignore
 
 === `max_bad_records`
 
-The maximum number of bad records that will be ignored when reading data.
+The maximum number of bad records that are ignored when reading data.
```
Comment (JakeSCahill) [inline modules/components/pages/outputs/gcp_bigquery.adoc:237]:
```suggestion
The maximum number of bad records to ignore when reading data and <<ignore_unknown_values, `ignore_unknown_values`>> is set to `true`.
```

File: modules/components/pages/outputs/gcp_bigquery.adoc [hunk 13]
```diff
@@ -225,8 +243,11 @@ The maximum number of bad records that will be ignored when reading data.
 
 === `auto_detect`
 
-Indicates if we should automatically infer the options and schema for CSV and JSON sources. If the table doesn't exist and this field is set to `false` the output may not be able to insert data and will throw insertion error. Be careful using this field since it delegates to the GCP BigQuery service the schema detection and values like `"no"` may be treated as booleans for the CSV format.
+Whether this component automatically infers the options and schema for `CSV` and `NEWLINE_DELIMITED_JSON` sources. 
+
+If this value is set to `false` and the destination table doesn't exist, the output may throw an insertion error as it is unable to insert data.
```
Comment (JakeSCahill) [inline modules/components/pages/outputs/gcp_bigquery.adoc:248]: Why 'may'? What other conditions determine whether an error is thrown?

File: modules/components/pages/outputs/gcp_bigquery.adoc [hunk 14]
```diff
@@ -225,8 +243,11 @@ The maximum number of bad records that will be ignored when reading data.
 
 === `auto_detect`
 
-Indicates if we should automatically infer the options and schema for CSV and JSON sources. If the table doesn't exist and this field is set to `false` the output may not be able to insert data and will throw insertion error. Be careful using this field since it delegates to the GCP BigQuery service the schema detection and values like `"no"` may be treated as booleans for the CSV format.
+Whether this component automatically infers the options and schema for `CSV` and `NEWLINE_DELIMITED_JSON` sources. 
+
+If this value is set to `false` and the destination table doesn't exist, the output may throw an insertion error as it is unable to insert data. 
 
+CAUTION: This field delegates schema detection to the GCP BigQuery service. For the `CSV` format, values like `"no"` may be treated as booleans.
```
Comment (JakeSCahill) [inline modules/components/pages/outputs/gcp_bigquery.adoc:250]:
```suggestion
CAUTION: This field delegates schema detection to the GCP BigQuery service. For the `CSV` format, values like `no` may be treated as booleans.
```

File: modules/components/partials/feature_stability_caution.adoc [hunk 15]
```diff
@@ -0,0 +1,12 @@
+ifndef::env-cloud[]
+[CAUTION]
+====
+This method is mostly stable. However, breaking changes could still be made to it outside of major version releases if a fundamental problem is found.
```
Comment (JakeSCahill) [inline modules/components/partials/feature_stability_caution.adoc:4]:
```suggestion
Breaking changes to this method may occur outside of major version releases only if a fundamental issue is identified.
```

File: modules/components/partials/feature_stability_caution.adoc [hunk 16]
```diff
@@ -0,0 +1,12 @@
...
+====
+endif::[]
+ifdef::env-cloud[]
+[CAUTION]
+====
+This method is mostly stable. However, breaking changes could still be made to it if a fundamental problem is found.
```
Comment (JakeSCahill) [inline modules/components/partials/feature_stability_caution.adoc:10]:
```suggestion
Breaking changes to this method may occur if a fundamental issue is identified.
```

File: modules/components/partials/feature_stability_caution.adoc (hunk 15 above)
Comment (JakeSCahill) [reply modules/components/partials/feature_stability_caution.adoc:4]: > This method is mostly stable.

This is too ambiguous and doesn't provide any useful information.

I added a suggestion but I still think it's weird. We're telling users that we aren't confident in the stability of the method and we won't necessarily follow the standard rules of semantic versioning if it breaks.

cc @Feediver1

File: modules/components/pages/inputs/nats.adoc (hunk 5 above)
Comment (asimms41) [reply modules/components/pages/inputs/nats.adoc:95]: Updated as requested.

File: modules/components/pages/inputs/redpanda_migrator.adoc (hunk 6 above)
Comment (asimms41) [reply modules/components/pages/inputs/redpanda_migrator.adoc:715]: Updated as requested. And yes that's right.

File: modules/components/pages/inputs/redpanda_migrator.adoc (hunk 8 above)
Comment (asimms41) [reply modules/components/pages/inputs/redpanda_migrator.adoc:732]: @mihaitodor. Under what circumstances would we recommend updating this option?

File: modules/components/pages/inputs/redpanda_migrator.adoc (hunk 7 above)
Comment (asimms41) [reply modules/components/pages/inputs/redpanda_migrator.adoc:713]: Updated as requested.

File: modules/components/pages/inputs/redpanda_migrator.adoc (hunk 9 above)
Comment (asimms41) [reply modules/components/pages/inputs/redpanda_migrator.adoc:721]: Updated as requested.

File: modules/components/pages/inputs/schema_registry.adoc (hunk 10 above)
Comment (asimms41) [reply modules/components/pages/inputs/schema_registry.adoc:133]: Updated as requested.

File: modules/components/pages/inputs/schema_registry.adoc (hunk 11 above)
Comment (asimms41) [reply modules/components/pages/inputs/schema_registry.adoc:403]: 1. Updated as requested.

File: modules/components/pages/outputs/gcp_bigquery.adoc (hunk 12 above)
Comment (asimms41) [reply modules/components/pages/outputs/gcp_bigquery.adoc:237]: Updated as requested.

File: modules/components/pages/outputs/gcp_bigquery.adoc (hunk 13 above)
Comment (asimms41) [reply modules/components/pages/outputs/gcp_bigquery.adoc:248]: Agreed. This error will occur. Updated to reflect this.

File: modules/components/pages/outputs/gcp_bigquery.adoc (hunk 14 above)
Comment (asimms41) [reply modules/components/pages/outputs/gcp_bigquery.adoc:250]: Updated as requested.

File: modules/components/partials/feature_stability_caution.adoc (hunk 15 above)
Comment (asimms41) [reply modules/components/partials/feature_stability_caution.adoc:4]: I've added the update for now. @Jeffail Is this just a legacy comment, or are breaking changes to Bloblang methods still released?

Comment (Jeffail) [reply modules/components/partials/feature_stability_caution.adoc]: We haven't made any breaking changes in the last few years but it's worth keeping the messaging there for now. We'd need to review each experimental component to say for sure whether it's okay to mark as stable.

File: modules/components/pages/inputs/redpanda_migrator.adoc (hunk 8 above)
Comment (mihaitodor) [reply modules/components/pages/inputs/redpanda_migrator.adoc:731]: I think some users want to have this system pick up newly created topics quicker. See https://github.com/redpanda-data/connect/issues/2879

File: modules/components/pages/inputs/redpanda_migrator.adoc (hunk 6 above)
Comment (mihaitodor) [reply modules/components/pages/inputs/redpanda_migrator.adoc]: Yep, that makes sense. I might add “ when creating copies of them in the destination”.

File: modules/components/pages/inputs/redpanda_migrator.adoc [hunk 17]
```diff
@@ -184,6 +187,31 @@ An optional consumer group. When specified, the partitions of specified topics a
 
 *Type*: `string`
 
+=== `output_resource`
+
+The label of the `redpanda_migrator` output in which the currently selected topics need to be created before attempting to read messages.
```
Comment (JakeSCahill) [inline modules/components/pages/inputs/redpanda_migrator.adoc:192]: Maybe add a link to the output docs and clarify that this is a separate connector.

Comment (asimms41) [reply modules/components/pages/inputs/redpanda_migrator.adoc:192]: Updated as requested on both the input and output of this component.

File: modules/components/pages/inputs/redpanda_migrator.adoc (hunk 8 above)
Comment (asimms41) [reply modules/components/pages/inputs/redpanda_migrator.adoc:731]: Updated the description to reflect this.

Comment (asimms41) [reply modules/components/pages/inputs/redpanda_migrator.adoc]: Again @mihaitodor. Could you have a quick check to make sure that I have understood this correctly.

File: modules/components/pages/inputs/redpanda_migrator.adoc (hunk 6 above)
Comment (mihaitodor) [reply modules/components/pages/inputs/redpanda_migrator.adoc]: LGTM, thank you!

**Review by JakeSCahill — APPROVED** [review]

**Pull request merged**